db/db_impl/db_impl.cc

```diff
@@ -660,6 +665,11 @@ Status DBImpl::CloseHelper() {
     }
   }
 
+  if (wbm_stall_) {
```

Comment:
What's the current behavior if writes are stalled when ```DB::Close()``` is called? Does it wait for all pending writes to finish?

Reply:
Oh.  I missed the removal of DBs from queue in case of DB::Close(). I will update the code.

Reply:
I couldn't write a test to know the exact behavior. But I have updated the code to remove DB from stall in case ```Close``` is called. Will that cover the scenario?. From the code it doesn't seem like it waits for pending writes to finish. It only waits for background jobs to finish. Do I need to handle this scenario or its upto users to not call shutdown in case there is a pending write.

Reply:
Right, I don't see the code waiting even for write controller initiated stalls. And ```Close()``` doesn't wait for in progress reads either. So it seems reasonable to assume that reads/writes must be quiesced before the DB is closed/destructed.